protocols/desktop-shell.xml

```diff
@@ -116,6 +116,32 @@
       </description>
       <arg name="position" type="uint"/>
     </request>
+    
+    <!-- Bar Hiding additions -->
+    <enum name="hide_modes">
+      <entry name="hide" value="0"/>
+      <entry name="show" value="1"/>
+    </enum>
+
+    <enum name="hide_state">
+      <entry name="hide" value="0"/>
+      <entry name="show" value="1"/>
+    </enum>
+    <request name="set_panel_hide_mode" since="3">
+      <description summary="set panel hide mode">
+        Tell the shell whether or not panel is in hide mode
+        i.e. whether to allocate a fixed amount of space
+        for the panel (hide mode false) or not (hide mode true)
+      </description>
+      <arg name="mode" type="uint"/>
+    </request>
+    
+    <request name="set_panel_hide" since="3">
+      <description summary="set if panel is currently hidden">
+        Tell the shell whether or not panel is currently hidden
+      </description>
+      <arg name="mode" type="uint"/>
+    </request>
```

Comment:
Hmm. 

Questions:
1. Is this the best API looking ahead? How will someone add support for other modes to this? how much work will it entail? 
2. Is this restricted to panels only? remember, this is a wayland protocol. The object could potentially be any window. It might be cool to be able to set hide_mode for arbitrary windows (someday...). Must this mode be coupled to the "placement" semantics of a "panel"?
3. Since the protocol supports enums. Is it necessary to tie the parameter to a specific mode? How about an API that takes a mode_id + param1/2/3/4. That's more extensible and keeps the protocol smaller.
3.